packages/proxy/lib/http/util/top-simulation.ts

```diff
@@ -0,0 +1,31 @@
+import { urlOriginsMatch, urlSameSiteMatch } from '@packages/network/lib/cors'
+import type { HttpMiddlewareThis } from '../index'
+import type { SecFetchSite } from '../../types'
+
+export const doesTopNeedToBeSimulated = <T>(ctx: HttpMiddlewareThis<T>): boolean => {
+  const currentAUTUrl = ctx.getAUTUrl()
+
+  // if the AUT url is undefined for whatever reason, return false as we do not want to complicate top simulation
+  if (!currentAUTUrl) {
+    return false
+  }
+
+  // only simulate top if the AUT is NOT the primary origin, meaning that we should treat the AUT as top
+  // or the request is the AUT frame, which is common for redirects and navigations.
+  const doesTopNeedToSimulating = !ctx.remoteStates.isPrimaryOrigin(currentAUTUrl) || ctx.req.isAUTFrame
+
+  return doesTopNeedToSimulating
+}
+
+// @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site
+export const calculateSiteContext = (url1: string, url2: string, isUserInteraction = false): SecFetchSite => {
+  if (urlOriginsMatch(url1, url2)) {
+    return 'same-origin'
+  }
+
+  if (urlSameSiteMatch(url1, url2)) {
+    return 'same-site'
+  }
+
+  return isUserInteraction ? 'none' : 'cross-site'
```

Comment:
How is 'isUserInteraction' set?

Reply:
It isn't. I believe I originally added it to cover https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site#directives `none` option, but this should only be an option if the user bookmarks, drag and drops a file in the browser window, or enters a url into the address bar, none of which should be widely applicable to use and we can remove it

Reply:
Do you need to check if it's cross-site vs cross-origin or does it matter?

Reply:
It matters in this case. A request can be `same-site` but also `cross-origin`

For instance, a request from `https://www.foobar.com`to `https://app.foobar.com` is a `same-site`/`cross-origin`, which if including credentials in this request would send first-party cookies. This is super confusing, so I put together a giant chart in the description of #23872 to showcase what gets sent and when.

Reply:
remove the userInteraction in 5a6765c6b6